### PR TITLE
Optimize chunk graph algorithm

### PR DESCRIPTION
**Pull request by sokra**
**What kind of change does this PR introduce?**
performance

**Did you add tests for your changes?**
existing tests

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
nothing

### CODE REVIEW

**Comment by sokra** [comment]
#8126

File: lib/Compilation.js [hunk 1]
```diff
@@ -1779,17 +1782,26 @@ class Compilation extends Tapable {
...
+		/** @type {Map<ChunkGroup, ChunkGroupInfo>} */
+		const chunkGroupInfoMap = new Map();
+
+		/** @type {Queue<ChunkGroup>} */
+		const queue2 = new Queue(inputChunkGroups);
+
+		for (const chunkGroup of inputChunkGroups) {
+			chunkGroupInfoMap.set(chunkGroup, {
+				minAvailableModules: undefined,
+				availableModulesToBeMerged: [new Set()]
```
Comment (ooflorent) [inline lib/Compilation.js]: Why does the array contain a `Set` and not on line 1912?

Comment (sokra) [reply lib/Compilation.js]: This handles the entrypoints which start with "no modules available".

line 1912 handles non-entrypoints which start uninitialized and available modules are inherited from parents.

**Comment by MLoughry** [comment]
I see ~12% shorter rebuild times in my repo. Definitely 👍

**Comment by sokra** [comment]
Great

**Pull request merged**